core/src/main/scala/com/softwaremill/diffx/DiffResult.scala

```diff
@@ -1,35 +1,42 @@
 package com.softwaremill.diffx
 import acyclic.skipped
+import DiffResult._
 
 trait DiffResult {
   def isIdentical: Boolean
 
-  def show: String = showIndented(5)
+  def show: String = showIndented(indentLevel)
 
   private[diffx] def showIndented(indent: Int): String
 }
 
+object DiffResult {
+  private[diffx] final val indentLevel = 5
+}
+
 case class DiffResultObject(name: String, fields: Map[String, DiffResult]) extends DiffResultDifferent {
   override private[diffx] def showIndented(indent: Int): String = {
-    val showFields = fields.map(f => s"${i(indent)}${f._1}: ${f._2.showIndented(indent + 5)}")
+    val showFields = fields.map(f => s"${i(indent)}${f._1}: ${f._2.showIndented(indent + indentLevel)}")
     s"""$name(
-       |${showFields.mkString("\n")})""".stripMargin
+       |${showFields.mkString(",\n")})""".stripMargin
   }
 }
 
 case class DiffResultMap(fields: Map[DiffResult, DiffResult]) extends DiffResultDifferent {
   override private[diffx] def showIndented(indent: Int): String = {
-    val showFields = fields.map(f => s"${i(indent)}${f._1.showIndented(indent + 5)}: ${f._2.showIndented(indent + 5)}")
+    val showFields =
+      fields.map(f =>
+        s"${i(indent)}${f._1.showIndented(indent + indentLevel)}: ${f._2.showIndented(indent + indentLevel)}"
+      )
     s"""Map(
-       |${showFields.mkString("\n")})""".stripMargin
+       |${showFields.mkString(",\n")})""".stripMargin
   }
 }
 
 case class DiffResultSet(diffs: List[DiffResult]) extends DiffResultDifferent {
   override private[diffx] def showIndented(indent: Int): String = {
-    val showFields = diffs.map(f => s"${i(indent)}$f")
```

Comment:
`$f` there was being `.toString`'d instead of calling `.showIndent`. Nested structures render with the diffx types wrapping the values.